### PR TITLE
Use validate_cmd & expose filename parameter

### PR DESCRIPTION
**Pull request by ekohl**
See individual commits for details.

### CODE REVIEW

File: manifests/java.pp [hunk 1]
```diff
@@ -29,49 +32,31 @@
...
-    returns     => 0,
-    refreshonly => true,
+  file { $filename:
+    ensure       => 'file',
+    source       => $source,
+    content      => $content,
+    mode         => '0644',
+    owner        => 'root',
+    group        => 'root',
+    validate_cmd => '/usr/bin/openssl x509 -in %s -noout',
```
Comment (bastelfreak) [inline manifests/java.pp:54]: are you sure that this path is correct on all operating systems? Is there something around that uses /bin/openssl? We had problems like that in the past.

Comment (ekohl) [reply manifests/java.pp:54]: Pretty sure. At least on EL (7+) `/bin` is a symlink to `/usr/bin`. Debian and Ubuntu both have it in `/usr/bin`. Not sure about suse mentioned in `metadata.json`, but I'd be quite confident in assuming it.

I wasn't sure if you can simply use `openssl` and expect it to use `$path`.

Comment (bastelfreak) [reply manifests/java.pp:54]: as far as I know it needs to be an absolute path, that's how we implemented it in other modules as well (puppet/nginx, saz/ssh)

**Comment by TheKrizz** [comment]
tag v5.1.0 has a bug
validate_cmd => '/usr/bin/openssl x509 -in %s -noout', is wrong,
it should be 
validate_cmd => '/usr/bin/openssl x509 -in % -noout',

with %s all temporary files has a 's' at the end and the validate_cmd command fails.

**Comment by ekohl** [comment]
> with %s all temporary files has a 's' at the end and the validate_cmd command fails because the filenames doesn't match.

Could you submit a PR to fix that?

**Pull request merged**